desafio-10/cbcalves/c/turing.c

```diff
@@ -0,0 +1,280 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define INT_MAX 2147483647 // definição do maior int
+
+// Definições feitas para um melhor controle de alocação da memória
+#define MALLOC(var, type, size)                                \
```

Comment:
O uso de macros pra esse tipo de coisa é complicado. Torna o debugging complicado demais. Se for o caso, crie uma função que faça isso, mas testar no caso a caso não é tão má idéia (idem pro REALLOC e FREE)

Reply:
Só não entendi o problema no debugging.

Reply:
Macros devem sempre ser usados de forma limitada. É muito facil cometer erros ao usar esses macros, já que a expansão é simples. No final eles viram uma função, mas o compilador reclama de um erro na função que não existe, já que o source é um macro. Outro problema é que se voce estiver usando um debugger, o codigo real não "bate" como fonte. Toda vez que voce olha o fonte, ve o macro, mas o codigo gerado é diferente e passa por outros caminhos.

Mais uma: No caso o programa é simples e é fácil de abortar de qualquer lugar, mas normalmente é melhor ir retornando os erros das camadas inferiores até as camadas superiores. Em um sistema de produção voce raramente quer que uma função aborte o programa sem ter o conhecimento do "todo" entende? Por exemplo, voce pode tentar alocar memoria e falhar. No seu caso ele aborta. Num sistema de produção, ele poderia tentar fazer com "pedaços" de memória menor.

Reply:
Entendi a questão do abort, eu mudei as macros e deixei só a alocação.